### PR TITLE
second iteration identity revisions

### PR DESCRIPTION
**Pull request by christinaausley**
## Description

Based on comments in https://github.com/camunda/developer-experience/issues/64, and a general review of all Identity content in Self-Managed.

Per [this comment](https://github.com/camunda/developer-experience/issues/64#issuecomment-1655745374), the callout under step https://github.com/camunda/developer-experience/issues/5 has been removed.

> The "Permissions are managed in Identity" link at the bottom of [this page](https://docs.camunda.io/docs/components/tasklist/userguide/using-tasklist/) to Identity goes to the next version, regardless of what version of the docs is selected.

I don't think this exists anymore. I couldn't locate it, but let me know if I missed something.

> There is a [first steps](https://docs.camunda.io/docs/next/self-managed/identity/getting-started/install-identity/) page in the docs already which tries to get a user going, but it is not visible enough.

I believe this has already been moved up, but am I missing more details here?

Also related to https://github.com/camunda-cloud/identity/issues/1966, and I've added an admonition to address this.

@Ben-Sheppard also noted that we suggest now a more recent version of Keycloak (we now support Keycloak v21) but all the screenshots are for Keycloak v19. @Ben-Sheppard this should likely be addressed in a separate PR, but who is best to work with on this? Pushing me to revisit screenshot automation 😅 

## When should this change go live?

- [ ] This change is not yet live and should not be merged until {ADD_DATE} (apply `hold` label or convert to draft PR)?
- [x] There is no urgency with this change.
- [ ] This change or page is part of a marketing blog, conference talk, or something else on a schedule.
- [ ] This functionality is already available but undocumented.
- [ ] This is a bug fix or security concern.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-platform-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

File: docs/self-managed/identity/user-guide/configuration/configure-external-identity-provider.md [hunk 1]
```diff
@@ -32,7 +32,9 @@ To configure an external identity provider like OpenID Connect, SAML, LDAP, or A
       ![keycloak-add-user-federation](../img/keycloak-add-user-federation.png)
 
 :::tip
-Keycloak supports a wide variety of authentication options, such as mapping external user groups, roles, or scopes to internal roles, and configuring the login screen and flow when multiple providers are added. Visit the Keycloak documentation for details on [adding a provider](https://www.keycloak.org/docs/19.0.3/server_admin/index.html#adding-a-provider),
+Keycloak supports a wide variety of authentication options, such as mapping external user groups, roles, or scopes to internal roles, and configuring the login screen and flow when multiple providers are added.
+
+Visit the Keycloak documentation for details on [adding a provider](https://www.keycloak.org/docs/19.0.3/server_admin/index.html#adding-a-provider),
 [configuring authentication](https://www.keycloak.org/docs/19.0.3/server_admin/index.html#configuring-authentication), and
 [integrating identity providers](https://www.keycloak.org/docs/19.0.3/server_admin/index.html#_identity_broker).
 :::
```
Comment (akeller) [inline docs/self-managed/identity/user-guide/configuration/configure-external-identity-provider.md:40]: Based on the table in [Supported environments in 8.2](https://docs.camunda.io/docs/reference/supported-environments/#camunda-platform-8-self-managed), we support Keycloak 16.1.x, 18.x, and 19.x. Pointing to the most recent version we support makes sense.

However, in[ `/next/`](https://docs.camunda.io/docs/next/reference/supported-environments/#camunda-platform-8-self-managed) where these docs changes are happening in this PR, we mention supported for 21.x only. These links should be adjusted.

I don't see a great way to send people to Keycloak docs without a specific version in the URL. We could consider some variable per C8 version, but that would be out of scope of this PR.

Comment (akeller) [reply docs/self-managed/identity/user-guide/configuration/configure-external-identity-provider.md:40]: Other URLs in this PR would need to be adjusted too.

File: docs/self-managed/identity/user-guide/configuration/making-identity-production-ready.md [hunk 2]
```diff
@@ -2,23 +2,24 @@
...
 
 ### Backing up
 
-To ensure recovery is possible, we recommend regularly backing up the database that supports Keycloak.
+To ensure recovery is possible, we recommend regularly backing up the database supporting Keycloak.
 
 #### Helm deployment
 
-If you deployed the Camunda platform using our [Helm charts](../../../platform-deployment/helm-kubernetes/overview.md),
+If you deployed the Camunda Platform using our [Helm charts](../../../platform-deployment/helm-kubernetes/overview.md),
```
Comment (akeller) [inline docs/self-managed/identity/user-guide/configuration/making-identity-production-ready.md:22]:
```suggestion
If you deployed Camunda Platform 8 using our [Helm charts](../../../platform-deployment/helm-kubernetes/overview.md),
```

Nit pick.

**Review by akeller — CHANGES_REQUESTED** [review]
Can you have a look at the casing suggestions below? WDYT?

* Casing of Getting Started Guide -> getting started guide (what-is-identity.md)
* Casing of Installation Guide -> installation guide (install-identity)

## what-is-identity.md
If we consider users who onboarding to Camunda via SaaS or are moving from Camunda Platform 7 to Camunda Platform 8 Self-Managed, can we introduce them to Identity via similar concepts? For example, SaaS users will be familiar with handling authentication and authorization in Console. Are there similarities and differences we can add to the `what-is-identity.md` page to help them understand Identity and Camunda Platform 8 Self-Managed as quickly as possible?

## install-identity.md
Is `http://localhost:8080` the best place to send all users? As in, is this the default or recommended port? If using Helm/Kubernetes, do we have an idea for how the URL should look? Let's link to things like https://docs.camunda.io/docs/self-managed/platform-deployment/troubleshooting/#identity-contextpath.

## connect-to-an-existing-keycloak.md
In the **Considerations** section, this should be rewritten to remove "we". Who or what is "we"? 

## Misc
I'm not sure where the best place to mention this is, but we should probably distinguish between a client in C8 SaaS and a client in Identity. If I'm reading the docs top to bottom in the Identity section, I see mention of clients in connect-to-an-existing-keycloak.md first.

On the screenshots, we use a lot of "Test{something}". Instead could we use terms and names that someone might have in their production system? See https://github.com/camunda/camunda-platform-docs/issues/2046

(Note, it may not make sense to cover all this feedback in this PR, but I'm raising it here as I reviewed the content by running this branch locally.)

File: docs/self-managed/identity/user-guide/configuration/configure-external-identity-provider.md (hunk 1 above)
Comment (Ben-Sheppard) [reply docs/self-managed/identity/user-guide/configuration/configure-external-identity-provider.md:40]: For clarity, we also support Keycloak v21 in the 8.2 branch now however this needs to be added to the matrix. I have created a separate issue for this https://github.com/camunda-cloud/identity/issues/1976

File: docs/self-managed/identity/deployment/starting-configuration.md [hunk 3]
```diff
@@ -24,25 +25,25 @@ create or update the following entities in Keycloak:
...
+| Name        | Created/updated with component |
 | ----------- | ------------------------------ |
 | Identity    | All                            |
 | Operate     | Operate                        |
 | Optimize    | Optimize                       |
 | Tasklist    | Tasklist                       |
 | Web Modeler | Web Modeler                    |
 
-### Client Scopes
+### Client ccopes
```
Comment (Ben-Sheppard) [inline docs/self-managed/identity/deployment/starting-configuration.md:36]:
```suggestion
### Client scopes
```

File: docs/self-managed/identity/troubleshooting/common-problems.md [hunk 4]
```diff
@@ -2,11 +2,12 @@
 id: common-problems
 title: "Common problems"
 sidebar_label: "Common problems"
+description: "Learn about solutions if Identity is unable to connect to Keycloak, or having issues getting Keycloak started."
```
Comment (Ben-Sheppard) [inline docs/self-managed/identity/troubleshooting/common-problems.md:5]: Here the descriptions are pretty specific, could we generalise this so the description remains fully relevant to the future content? Something like:
```
Learn about solutions to common issues in Identity
```

File: docs/self-managed/identity/troubleshooting/common-problems.md [hunk 5]
```diff
@@ -30,26 +33,24 @@ Keycloak is ready to accept connections when the following log lines are visible
...
+When the Keycloak service is ready for connections, start (or restart) the Identity pod.
 
 ### Solution 2: Identity making requests from an external IP address
 
 By default, Keycloak requires TLS on requests that originate from what it considers to be an external source. The Keycloak
 documentation for [setting up SSL](https://www.keycloak.org/docs/19.0.3/server_installation/#_setting_up_ssl) maintains
 a list of what they consider to be an external IP address under the `external requests` section.
 
-The solution to this issue will depend largely on your environment, however as a starting point we would suggest you consider
+The solution to this issue will depend largely on your environment.However, as a starting point we would suggest you consider
```
Comment (Ben-Sheppard) [inline docs/self-managed/identity/troubleshooting/common-problems.md:44]:
```suggestion
The solution to this issue will depend largely on your environment. However, as a starting point we would suggest you consider
```

File: docs/self-managed/identity/troubleshooting/troubleshoot-identity.md [hunk 6]
```diff
@@ -2,6 +2,7 @@
 id: troubleshoot-identity
 title: "Troubleshooting Identity"
 sidebar_label: "Overview"
+description: "Learn how to resolve issues if the Identity pod crashloops or continually restarts, and more."
```
Comment (Ben-Sheppard) [inline docs/self-managed/identity/troubleshooting/troubleshoot-identity.md:5]: Ah, I like the `and more` here :)

File: versioned_docs/version-8.2/self-managed/identity/deployment/starting-configuration.md [hunk 7]
```diff
@@ -24,25 +25,25 @@ create or update the following entities in Keycloak:
...
+| Name        | Created/updated with component |
 | ----------- | ------------------------------ |
 | Identity    | All                            |
 | Operate     | Operate                        |
 | Optimize    | Optimize                       |
 | Tasklist    | Tasklist                       |
 | Web Modeler | Web Modeler                    |
 
-### Client Scopes
+### Client ccopes
```
Comment (Ben-Sheppard) [inline versioned_docs/version-8.2/self-managed/identity/deployment/starting-configuration.md:36]:
```suggestion
### Client scopes
```

File: versioned_docs/version-8.2/self-managed/identity/troubleshooting/common-problems.md [hunk 8]
```diff
@@ -30,26 +33,24 @@ Keycloak is ready to accept connections when the following log lines are visible
...
+When the Keycloak service is ready for connections, start (or restart) the Identity pod.
 
 ### Solution 2: Identity making requests from an external IP address
 
 By default, Keycloak requires TLS on requests that originate from what it considers to be an external source. The Keycloak
 documentation for [setting up SSL](https://www.keycloak.org/docs/19.0.3/server_installation/#_setting_up_ssl) maintains
 a list of what they consider to be an external IP address under the `external requests` section.
 
-The solution to this issue will depend largely on your environment, however as a starting point we would suggest you consider
+The solution to this issue will depend largely on your environment.However, as a starting point we would suggest you consider
```
Comment (Ben-Sheppard) [inline versioned_docs/version-8.2/self-managed/identity/troubleshooting/common-problems.md:44]:
```suggestion
The solution to this issue will depend largely on your environment. However, as a starting point we would suggest you consider
```

File: versioned_docs/version-8.2/self-managed/identity/user-guide/configuration/making-identity-production-ready.md [hunk 9]
```diff
@@ -2,23 +2,24 @@
...
 
 ### Backing up
 
-To ensure recovery is possible, we recommend regularly backing up the database that supports Keycloak.
+To ensure recovery is possible, we recommend regularly backing up the database supporting Keycloak.
 
 #### Helm deployment
 
-If you deployed the Camunda platform using our [Helm charts](../../../platform-deployment/helm-kubernetes/overview.md),
+If you deployed the Camunda Platform using our [Helm charts](../../../platform-deployment/helm-kubernetes/overview.md),
```
Comment (Ben-Sheppard) [inline versioned_docs/version-8.2/self-managed/identity/user-guide/configuration/making-identity-production-ready.md:22]:
```suggestion
If you deployed the Camunda Platform 8 using our [Helm charts](../../../platform-deployment/helm-kubernetes/overview.md),
```

File: docs/self-managed/identity/user-guide/configuration/configure-external-identity-provider.md (hunk 1 above)
Comment (akeller) [reply docs/self-managed/identity/user-guide/configuration/configure-external-identity-provider.md]: @Ben-Sheppard, for something like this where the direction is clear (adding v21 to 8.2), we can update the matrix and URLs, then add you as a reviewer. Negotiate with us! We are happy to help as we can.

File: versioned_docs/version-8.2/self-managed/identity/user-guide/configuration/making-identity-production-ready.md (hunk 9 above)
Comment (christinaausley) [reply versioned_docs/version-8.2/self-managed/identity/user-guide/configuration/making-identity-production-ready.md]: Resolved in latest commit 👍

File: versioned_docs/version-8.2/self-managed/identity/troubleshooting/common-problems.md (hunk 8 above)
Comment (christinaausley) [reply versioned_docs/version-8.2/self-managed/identity/troubleshooting/common-problems.md]: Resolved in latest commit 👍

File: versioned_docs/version-8.2/self-managed/identity/deployment/starting-configuration.md (hunk 7 above)
Comment (christinaausley) [reply versioned_docs/version-8.2/self-managed/identity/deployment/starting-configuration.md]: Resolved in latest commit 👍

File: docs/self-managed/identity/troubleshooting/common-problems.md (hunk 5 above)
Comment (christinaausley) [reply docs/self-managed/identity/troubleshooting/common-problems.md]: Resolved in latest commit 👍

File: docs/self-managed/identity/troubleshooting/common-problems.md (hunk 4 above)
Comment (christinaausley) [reply docs/self-managed/identity/troubleshooting/common-problems.md]: Resolved in latest commit 👍

File: docs/self-managed/identity/deployment/starting-configuration.md (hunk 3 above)
Comment (christinaausley) [reply docs/self-managed/identity/deployment/starting-configuration.md]: Resolved in latest commit 👍

**Comment by christinaausley** [comment]
> Can you have a look at the casing suggestions below? WDYT?

Adjusted in latest commit.

> If we consider users who onboarding to Camunda via SaaS or are moving from Camunda Platform 7 to Camunda Platform 8 Self-Managed, can we introduce them to Identity via similar concepts? For example, SaaS users will be familiar with handling authentication and authorization in Console. Are there similarities and differences we can add to the what-is-identity.md page to help them understand Identity and Camunda Platform 8 Self-Managed as quickly as possible?

Do you have thoughts here before I make any adjustments @Ben-Sheppard?

> Is `http://localhost:8080` the best place to send all users? As in, is this the default or recommended port? If using Helm/Kubernetes, do we have an idea for how the URL should look? Let's link to things like https://docs.camunda.io/docs/self-managed/platform-deployment/troubleshooting/#identity-contextpath.

Also curious on your thoughts here @Ben-Sheppard.

> In the Considerations section, this should be rewritten to remove "we". Who or what is "we"?

Adjusted in latest commit.

> I'm not sure where the best place to mention this is, but we should probably distinguish between a client in C8 SaaS and a client in Identity. If I'm reading the docs top to bottom in the Identity section, I see mention of clients in connect-to-an-existing-keycloak.md first.

> On the screenshots, we use a lot of "Test{something}". Instead could we use terms and names that someone might have in their production system? See https://github.com/camunda/camunda-platform-docs/issues/2046

I'll likely create separate issues for these. Thoughts @Ben-Sheppard?

**Pull request merged**